pkg/cmd/pr/shared/survey.go

```diff
@@ -345,19 +345,34 @@ func MetadataSurvey(io *iostreams.IOStreams, client *api.Client, baseRepo ghrepo
 			fmt.Fprintln(io.ErrOut, "warning: no milestones in the repository")
 		}
 	}
-	values := metadataValues{}
-	err = prompt.SurveyAsk(mqs, &values, survey.WithKeepFilter(true))
-	if err != nil {
-		return fmt.Errorf("could not prompt: %w", err)
-	}
-	state.Reviewers = values.Reviewers
-	state.Assignees = values.Assignees
-	state.Labels = values.Labels
-	state.Projects = values.Projects
-	if values.Milestone != "" && values.Milestone != noMilestone {
-		state.Milestones = []string{values.Milestone}
+
+	if len(mqs) > 0 {
+		values := metadataValues{
+			Reviewers: state.Reviewers,
+			Assignees: state.Assignees,
+			Labels:    state.Labels,
+			Projects:  state.Projects,
+		}
+		if len(state.Milestones) > 0 {
+			values.Milestone = state.Milestones[0]
+		}
+
+		err = prompt.SurveyAsk(mqs, &values, survey.WithKeepFilter(true))
+		if err != nil {
+			return fmt.Errorf("could not prompt: %w", err)
+		}
+
+		state.Reviewers = values.Reviewers
+		state.Assignees = values.Assignees
+		state.Labels = values.Labels
+		state.Projects = values.Projects
+		if values.Milestone != "" && values.Milestone != noMilestone {
+			state.Milestones = []string{values.Milestone}
+		}
 	}
 
+	state.MetadataResult = nil
```

Comment:
At first I wasn't clear why you did this, but then I tested it out and noticed that it fails in `AddMetadataToIssueParams` without this line.

I'm not convinced of this workaround, though, because it clears the cache of what was just fetched from the server and causes a whole bunch of records to be fetched again. I realize that fixing this properly is non-trivial, and I don't require that you add it to your PR. I'm currently exploring alternatives.